frontend/public/module/k8s/cluster-settings.ts

```diff
@@ -256,3 +258,37 @@ export const getConditionUpgradeableFalse = (resource) =>
 
 export const getNotUpgradeableResources = (resources) =>
   resources.filter((resource) => getConditionUpgradeableFalse(resource));
+
+export enum NodeTypes {
+  master = 'master',
+  worker = 'worker',
+}
+
+export enum NodeTypeNames {
+  Master = 'Master',
+  Worker = 'Worker',
+}
+
+export const isMCPMaster = (mcp: MachineConfigPoolKind) => mcp.metadata.name === NodeTypes.master;
+
+export const isMCPWorker = (mcp: MachineConfigPoolKind) => mcp.metadata.name === NodeTypes.worker;
+
+export const isMCPPaused = (mcp: MachineConfigPoolKind) => mcp.spec.paused;
+
+export const sortMCPsByCreationTimestamp = (a: MachineConfigPoolKind, b: MachineConfigPoolKind) =>
+  a.metadata.creationTimestamp.localeCompare(b.metadata.creationTimestamp);
+
+export const clusterIsUpToDateOrUpdateAvailable = (status: ClusterUpdateStatus) =>
+  status === ClusterUpdateStatus.UpToDate || status === ClusterUpdateStatus.UpdatesAvailable;
+
+export const getMCPsToPausePromises = (
+  machineConfigPools: MachineConfigPoolKind[],
+  paused: boolean,
+) => {
+  const promises = [];
+  machineConfigPools.forEach((mcp) => {
+    const patch = [{ op: 'add', path: '/spec/paused', value: paused }];
+    promises.push(k8sPatch(MachineConfigPoolModel, mcp, patch));
+  });
+  return promises;
```

Comment:
Ah, haha... so sorry. This is my fault for trying to explain everything and not being clear about what I was wanting.

`.map` is fine, but you need to return out of the predicate. Not mutating is better than mutating. So your current state really should be a middle ground between what you had and what you have now. Again, this is my bad for explaining poorly.

```suggestion
  return machineConfigPools.map((mcp) => {
    const patch = [{ op: 'add', path: '/spec/paused', value: paused }];
    return k8sPatch(MachineConfigPoolModel, mcp, patch);
  });
```